server/src/main/java/org/apache/iotdb/db/engine/settle/SettleRequestHandler.java

```diff
@@ -0,0 +1,228 @@
+      }
+
+      if (TsFileUtils.isSequence(tsFile)) {
+        hasSeqFile = true;
+      } else {
+        hasUnSeqFile = true;
+      }
+      tsFileNames.add(fileName);
+      if (hasSeqFile && hasUnSeqFile) {
+        return RpcUtils.getStatus(
+            TSStatusCode.UNSUPPORTED_OPERATION, "Settle by cross compaction is not allowed.");
+      }
+    }
+
+    if (!hasModsFile) {
+      return RpcUtils.getStatus(
+          TSStatusCode.ILLEGAL_PARAMETER, "At least one mods file should be selected");
+    }
+    if (dataRegion == null) {
+      return RpcUtils.getStatus(TSStatusCode.ILLEGAL_PATH, "DataRegion not exist");
+    }
+    TsFileManager tsFileManager = dataRegion.getTsFileManager();
+    if (!tsFileManager.isAllowCompaction()) {
+      return RpcUtils.getStatus(
+          TSStatusCode.UNSUPPORTED_OPERATION, "Compaction in this DataRegion is not allowed.");
+    }
+
+    // try to get a continuous TsFileResource list with input TsFile names
+    List<TsFileResource> tsFileResources =
+        getContinuousTsFileResourcesByFileNames(
+            tsFileManager, timePartitionId, hasSeqFile, tsFileNames);
+    if (tsFileResources.size() != tsFileNames.size()) {
+      return RpcUtils.getStatus(
+          TSStatusCode.ILLEGAL_PARAMETER, "Could not find enough satisfied TsFile");
+    }
+
+    IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+    if (hasSeqFile && !config.isEnableSeqSpaceCompaction()) {
+      return RpcUtils.getStatus(
+          TSStatusCode.UNSUPPORTED_OPERATION, "Compaction in Seq Space is not enabled");
+    }
+    if (hasUnSeqFile && !config.isEnableUnseqSpaceCompaction()) {
+      return RpcUtils.getStatus(
+          TSStatusCode.UNSUPPORTED_OPERATION, "Compaction in Unseq Space is not enabled");
+    }
+    if (tsFileResources.size() > config.getMaxInnerCompactionCandidateFileNum()) {
+      return RpcUtils.getStatus(TSStatusCode.UNSUPPORTED_OPERATION, "File nums is too much.");
+    }
+    if (testMode) {
+      return RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
+    }
+
+    AbstractCompactionTask task =
+        new InnerSpaceCompactionTask(
+            timePartitionId,
+            tsFileManager,
+            tsFileResources,
+            hasSeqFile,
+            config.getInnerSeqCompactionPerformer().createInstance(),
+            CompactionTaskManager.currentTaskNum,
+            tsFileManager.getNextCompactionTaskId());
+    try {
+      CompactionTaskManager.getInstance().addTaskToWaitingQueue(task);
+    } catch (InterruptedException e) {
+      logger.error(
+          "meet error when adding task-{} to compaction waiting queue", task.getSerialId());
+      return RpcUtils.getStatus(
+          TSStatusCode.COMPACTION_ERROR, "meet error when submit settle task.");
+    }
+    return RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
+  }
+
+  private List<TsFileResource> getContinuousTsFileResourcesByFileNames(
```

Comment:
Actually what you want to do are two things:
1. validate whether the selected files are continuous
2. If all the validation are passed, get the corresponding TsFileResource.

Do not merge these two functionalities into one method because it will make the code un-readable.